scripts/pythondistdeps.py

```diff
@@ -130,200 +134,289 @@ def convert_ordered(name, operator, version_id):
-            else:
-                warn("Version for {!r} has not been found".format(dist), RuntimeWarning)
-                continue
-        if args.majorver_provides or args.majorver_only or args.legacy_provides or args.legacy:
-            # Get the Python major version
-            pyver_major = dist.py_version.split('.')[0]
-        if args.provides:
-            # If egg/dist metadata says package name is python, we provide python(abi)
-            if dist.key == 'python':
-                name = 'python(abi)'
-                if name not in py_deps:
-                    py_deps[name] = []
-                py_deps[name].append(('==', dist.py_version))
-            if not args.legacy or not args.majorver_only:
-                name = 'python{}dist({})'.format(dist.py_version, dist.key)
-                if name not in py_deps:
-                    py_deps[name] = []
-            if args.majorver_provides or args.majorver_only:
-                pymajor_name = 'python{}dist({})'.format(pyver_major, dist.key)
-                if pymajor_name not in py_deps:
-                    py_deps[pymajor_name] = []
-            if args.legacy or args.legacy_provides:
-                legacy_name = 'pythonegg({})({})'.format(pyver_major, dist.key)
-                if legacy_name not in py_deps:
-                    py_deps[legacy_name] = []
-            if dist.version:
-                version = dist.version
-                spec = ('==', version)
-                if spec not in py_deps[name]:
-                    if not args.legacy:
+    try:
+        return OPERATORS[operator](name, operator, version_id)
+    except Exception as exc:
+        raise RuntimeError("Cannot process Python package version `{}` for name `{}`".
+                           format(version_id, name)) from exc
+
+
+def normalize_name(name):
+    """https://www.python.org/dev/peps/pep-0503/#normalized-names"""
+    import re
+    return re.sub(r'[-_.]+', '-', name).lower()
+
+
+if __name__ == "__main__":
+    """To allow this script to be importable (and its classes/functions
+       reused), actions are performed only when run as a main script."""
+
+    parser = argparse.ArgumentParser(prog=argv[0])
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('-P', '--provides', action='store_true', help='Print Provides')
+    group.add_argument('-R', '--requires', action='store_true', help='Print Requires')
+    group.add_argument('-r', '--recommends', action='store_true', help='Print Recommends')
+    group.add_argument('-C', '--conflicts', action='store_true', help='Print Conflicts')
+    group.add_argument('-E', '--extras', action='store_true', help='Print Extras')
+    group_majorver = parser.add_mutually_exclusive_group()
+    group_majorver.add_argument('-M', '--majorver-provides', action='store_true', help='Print extra Provides with Python major version only')
+    group_majorver.add_argument('--majorver-provides-versions', action='store',
```

Comment:
Instead of making this comma separated, can we use `append` or `extend` instead? It would make sense to store internally as a list. This would eliminate the weird formatting requirements too...

Reply:
`append` shouldn't be a problem. `extend` however, swallows unlimited number of arguments, which could interfere with positional arguments, so I'd rather avoid it. How about using `append` and also keeping the option for comma-separated entries?

Reply:
Do we need comma-separated entries if we use `append`? Most people would call it at most twice.

Reply:
I agree that we don't *need* it. How big difference it is if we *support* it?

Reply:
I'm mostly thinking of potential field parsing bugs. I'm thinking of if people call it twice with two comma separated lists. I don't know if we really want to make that case possible...

Reply:
> I'm mostly thinking of potential field parsing bugs. I'm thinking of if people call it twice with two comma separated lists. I don't know if we really want to make that case possible...

The reason I'm suggesting it is that `--majorver-provides-versions` doesn't have a short form (and it's so specific that I don't think it should), and it would result in very long calls, e.g. in Fedora we would need:

    --provides --majorver-provides-versions 2.7 --majorver-provides-versions 3.8 --normalized-names-format pep503 --normalized-names-provide-both

I can actually make a test for all the possible ways to call `--majorver-provides-versions` so it should be safe to implement it both ways.

Reply:
Hmm, okay 👍